src/fragment-components/a-date-picker.tsx

```diff
@@ -0,0 +1,412 @@
+				labelText='Invalid'
+				id='invalid'
+				checked={selectedComponent.invalid}
+				onChange={(checked: boolean) => setComponent({
+					...selectedComponent,
+					invalid: checked
+			})} />
+			<TextInput
+				value={selectedComponent.rangeStartLabel}
+				labelText='Date picker label'
+				onChange={(event: any) => setComponent({
+					...selectedComponent,
+					rangeStartLabel: event.currentTarget.value
+			})} />
+			{
+				selectedComponent.datePickerType === 'range'
+					&& <TextInput
+					value={selectedComponent.rangeEndLabel}
+					labelText='Date picker label'
+					onChange={(event: any) => setComponent({
+						...selectedComponent,
+						rangeEndLabel: event.currentTarget.value
+					})} />
+			}
+				</> :
+			<DraggableTileList
+				dataList={[...selectedComponent.items]}
+				setDataList={updateStepList}
+				updateItem={updateListItems}
+				defaultObject={{
+					text: 'New timezone',
+					value: 'New timezoneValue'
+				}}
+				template={template} />
+		}
+	</>;
+};
+
+export const ADatePickerCodeUI = ({ selectedComponent, setComponent }: any) => <TextInput
+	value={selectedComponent.codeContext?.name}
+	labelText='Input name'
+	onChange={(event: any) => setComponent({
+		...selectedComponent,
+		codeContext: {
+			...selectedComponent.codeContext,
+			name: event.currentTarget.value
+		}
+	})}
+/>;
+
+export const ADatePicker = ({
+	componentObj,
+	...rest
+}: any) => {
+	return (
+		<AComponent
+		componentObj={componentObj}
+		rejectDrop={true}
+		{...rest}>
+			{
+				componentObj.datePickerType === 'timePicker'
+					? <TimePicker id="time-picker" light={componentObj.light}>
+						<TimePickerSelect labelText="time-picker-1" id="time-picker-select-1">
+							<SelectItem value="AM" text="AM" />
+							<SelectItem value="PM" text="PM" />
+						</TimePickerSelect>
+						<TimePickerSelect labelText="time-picker-2" id="time-picker-select-2" >
+							{
+								componentObj.items.map((step: any, index: number) => <SelectItem
+									value={step.value}
+									text={step.text}
+									key={index}
+								/>)
+							}
+						</TimePickerSelect>
+					</TimePicker>
+					: <DatePicker
+					dateFormat={componentObj.dateFormat}
+					datePickerType={componentObj.datePickerType}
+					light={componentObj.light}>
+						<DatePickerInput
+							id="date-picker-default-id"
+							placeholder={componentObj.placeHolder}
+							labelText={componentObj.rangeStartLabel}
+							type="text"
+							size={componentObj.size}
+							disabled={componentObj.disabled}
+							invalid={componentObj.invalid} />
+						{
+							componentObj.datePickerType === 'range' &&
+								<DatePickerInput
+									id="date-picker-range-end"
+									placeholder={componentObj.placeHolder}
+									labelText={componentObj.rangeEndLabel}
+									type="text"
+									size={componentObj.size}
+									disabled={componentObj.disabled}
+									invalid={componentObj.invalid} />
+						}
+					</DatePicker>
+			}
+		</AComponent>
```

Comment:
Why is time picker in this PR??? Time Picker & date picker are different components

Reply:
its one component no? just a different varient. Similar to this https://carbondesignsystem.com/components/date-picker/usage/

Reply:
They are grouped but I wouldn't consider them as the same component. @zvonimirfras Thoughts on this? Combining all the variants into a single component results in the JSON export being bloated - obviously it won't be noticeable but still very unclean.

![image](https://user-images.githubusercontent.com/38994122/206343699-d60b8f31-8fb1-4204-8553-d2eb5fb15731.png)

Reply:
Its one component in carbon with different variants. I though we would keep it the same to avoid any confusion for people referring the carbon docs.

Reply:
I disagree, they are not variants of the same component. They are just grouped together here...

One is a calendar which will return a date in ISO format and the other is a `text input` that accepts time & will return a string & the string, 2 time periods (AM/PM), & time zones

Reply:
Its all grouped together because in the Carbon Components Design components there is no Time Picker component.. Doesn't it make sense to carry the same pattern here if we're using carbon design system as a base guideline.